### DISCUSSION TITLE
How to run from PowerShell 7 on macOS (or Linux)

### DISCUSSION THREAD

**Discussion by JunielKatarn** [dlemstra/Magick.NET#665] [Help] [answered]
Got a very simple program that uses Magick.NET and works correctly:
```
using System;

using ImageMagick;

namespace Command
{
    class Program
    {
        static void Main(string[] args)
        {
            if (args.Length != 2)
                return;

            using (var svg = new MagickImage(args[0], MagickFormat.Svg))
            {
                svg.Resize(724, 1024);
                svg.Write(args[1], MagickFormat.Png);
            }
        }
    }
}

```

I'd like to script it using PowerShell, but I'm apparently not following the right steps to load the native assemblies. Note, I'm trying to load them from the installed NuGet packages:
```
Add-Type -Path ./magick.net.core/2.0.0/lib/netstandard20/Magick.NET.Core.dll
Add-Type -Path ./magick.net-q8-anycpu/7.19.0/lib/netstandard20/Magick.NET-Q8-AnyCPU.dll
```

PowerShell does show ImageMagick types in autocomplete, but fails to run the same code as the C# program:
```
> $svg = [ImageMagick.MagickImage]::new( "/path/to/my.svg", [ImageMagick.MagickFormat]::Svg )
MethodInvocationException: Exception calling ".ctor" with "2" argument(s): "The type initializer for 'NativeMagickSettings' threw an exception."
```

Can't see the actual exception, but my guess is PowerShell is not finding the native library `magick.net-q8-anycpu/7.19.0/runtimes/osx-x64/native/Magick.Native-Q8-x64.dll.dylib`.

Any advice?

Thanks.

**Comment by dlemstra** [comment, marked as answer]
I think it would work if you copy those three files to the same folder and the run Add-Type from that folder.

**Reply by JunielKatarn** [reply]
Just placing the DYLD next to the Q8-AnyCPU DLL worked.
Even a symlink (full, not relative path) does the trick :)

That said, I want to find a way to tell the PowerShell .NET instance to use a specific path to look for the DYLD (`DYLD_LIBRARY_PATH` and `DYLD_FALLBACK_LIBRARY_PATH` haven't worked so far), but that's an unrelated topic.

Thanks!
